arangod/Agency/Agent.cpp

```diff
@@ -425,7 +459,11 @@ priv_rpc_ret_t Agent::recvAppendEntriesRPC(term_t term, std::string const& leade
           << " with term " << term;
       {
         WRITE_LOCKER(oLocker, _outputLock);
-        _commitIndex = std::max(_commitIndex, std::min(leaderCommitIndex, lastIndex));
+        auto const tmp = std::max(_commitIndex, std::min(leaderCommitIndex, lastIndex));
```

Comment:
`index_t` is only mildly longer than `auto`, therefore I would prefer it.
```suggestion
        index_t const tmp = std::max(_commitIndex, std::min(leaderCommitIndex, lastIndex));
```